### DISCUSSION TITLE
Sync Users with Microsoft Entra ID

### DISCUSSION THREAD

**Discussion by Ashkhatoyant1a** [dani-garcia/vaultwarden#4550] [Q&A]
Hey there, I want to Sync my users in Entra ID with Vaultwarden. I know that Vaultwarden haven't builtin SSO. PLease can you help? Has anybody done something like that? I have an Idea to use an API to get users information but I can't find how I can PUT it on my Vaultwarden Directory. I tought about Vaultwarden Ldap as well but I need a solution witch deleting users as well when they are deleted in Entra ID (I can make it via API using tools like Zappier or n8n)

**Comment by jobritz** [comment]
https://github.com/Timshel/vaultwarden Check out this repo

**Comment by jobritz** [comment]
I just noticed that you can also sync users with the Bitwarden Directory Connector from Entra ID. This connecters can remove accounts from an organization. Deleting an account is done over the Admin-Panel but I think you have to do it manually.

**Comment by ToniM76** [comment]
Hello,
is the SSO with Azure working?
I use the Proxmox Helper Scripts to install vaultwarden. 
What are the next steps to configure SSO
I allready created a Azure App Registration, and try to use this settings in /op/vaultwarden/.env
```
SSO_ENABLED=true
SSO_PROVIDER=azure
SSO_CLIENT_ID=<myid>
SSO_CLIENT_SECRET=<mySecret>
SSO_AUTHORITY=https://login.microsoftonline.com/<myID>/v2.0/.well-known/openid-configuration
SSO_REDIRECT_URI=https://<mydomain>/oidc/callback

```
But with this Settings nothing change on the Login, and there is also nothing in the Admin Panel to setup?

**Reply by jobritz** [reply]
SSO is not yet implemented in the official image. If you want to test it you need to use the image as described [here](https://github.com/Timshel/vaultwarden). Then you will also see a section in the admin panel

**Reply by PPCM** [reply]
Hello,
Do you know when it will be implemented in the official image?

**Comment by astrakid** [comment]
i successfully deployed timeshel-vaultwarden with entra sso. but this is not a sync, but an sso solution.
is there any way to sync users to vaultwarden? the problem is, that everyone would need to register in vaultwarden and then use sso login. we just want to only allow sso via entra. but this needs to set 
SSO_SIGNUPS_MATCH_EMAIL=true
SSO_ALLOW_UNKNOWN_EMAIL_VERIFICATION=true
this may lead to an account takeover, if someone gets access to an emailadress (which might happen due to same mailaddresses because of same name - we had this 3 times in the last 4 years). this is a security violation, which i am not able to solve.
any ideas?

**Reply by whoisbenbrown** [reply]
That sounds more like an issue with your user naming and mailbox retention policies. Same as AD objects, there is very little to gain from deleting them over moving them to a disabled/inactive state and sorting them to a disabled OU or similar. Important to note that Vaultwarden isn't the only credential reuse attack surface, so there is a reason to urge your organization to adopt these policies.
Microsoft [recommends this as a best practice.](https://learn.microsoft.com/en-us/purview/create-and-manage-inactive-mailboxes?view=o365-worldwide)
